kernel/xpmem_mmu_notifier.c

```diff
@@ -123,19 +117,17 @@ xpmem_invalidate_range(struct mmu_notifier *mn,
 		 * kernel can't rearrange the address space while a MMU notifier
 		 * callout is occurring.
 		 */
-		if (xpmem_is_vm_ops_set(vma))
+		if (!xpmem_is_vm_ops_set(vma))
 			continue;
 
-		vm_start = max(vma->vm_start, start);
-		if (vm_start >= vma->vm_end)
-			continue;
-
-		vm_end = min(vma->vm_end, end);
-		if (vm_end <= vma->vm_start)
-			continue;
-
-		xpmem_invalidate_PTEs_range(seg_tg, vm_start, vm_end);
+		vm_end = min(vma->vm_start, end);
```

Comment:
maybe add comment like
"if found an xpmem-attached VMA, invalidate the range before its start, and position 'curr' after its end'